presto-main/src/main/java/io/prestosql/operator/aggregation/builder/MergingHashAggregationBuilder.java

```diff
@@ -149,7 +150,7 @@ private void rebuildHashAggregationBuilder()
                 Optional.of(DataSize.succinctBytes(0)),
                 Optional.of(overwriteIntermediateChannelOffset),
                 joinCompiler,
-                false,
-                false);
+                // TODO: merging should also yield on memory reservations
+                () -> true);
```

Comment:
maybe
```
bytes -> {
     // We report hashAggregationBuilder's memory directly.
     // TODO: merging should also yield on memory reservations
     return true;
}
```

Reply:
> `// We report hashAggregationBuilder's memory directly.`

That's always the case as `updateMemory` just triggers that update